providers/download_cert.rb

```diff
@@ -0,0 +1,86 @@
+# Cookbook Name:: abiquo
+# Provider:: download_cert
+#
+# Copyright 2014, Abiquo
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+require 'uri'
+
+def whyrun_supported?
+    true
+end
+
+use_inline_resources
+
+action :download do
+    converge_by("Downloading SSL certificate from #{new_resource.host}") do
+        # Check the provided host is not nil.
+        # This can happen if the node has no property for
+        # abiquo.server.api.location
+        if new_resource.host.nil?
+            Chef::Log.debug 'abiquo_download_cert :: SSL API location not provided. Skipping.'
```

Comment:
Same here. This is not only for api location properties.
BTW, this check could be removed if we apply the suggested change to move the api location host to the list of additional hosts.